nntrainer/utils/meson.build

```diff
@@ -24,9 +24,13 @@ if get_option('enable-trace')
   util_headers += 'tracer.h'
 endif
 
-if get_option('enable-fp16')
+arch = target_machine.cpu_family()
+
+if get_option('platform') == 'android' or arch == 'aarch64'
```

Comment:
Nope. If it's aarch64 with armv8.0, you cannot use fp16 simd-neon.
You need to depend on fp16 enable/disable explicitly

Reply:
Then how about:
1. fp16 using and android -> use neon
```meson
if get_option('enable-fp16') && get_option('platform') == 'android'
```
or,

2. make explicit option for using neon
```meson
if get_option('enable-neon')
```
?

Current option would include neon files for linux build as well. So we need a new option to separate them.

Reply:
### Update 
During android build using `nntrainer/tools/package_android.sh` , 
`target_machine.cpu_family()` output is `x86_64` since we actually build on PC at this moment.
Thus think I should make a explicit option like:
```meson
if get_option('enable-neon')
```
and make such option true for ARM build.  

We were free from this problem because we just used 
```meson
if get_option('platform') == 'anrdoid'
```
which is not really a valid option.

This way, I think we can segregate the problem like:

$$ARM > NEON > FP16$$
> fyi, neon supports multiple dataTypes, so it is a bigger set than `ENABLE_FP16` option